In [1]:
import pandas as pd
import numpy as np

In [2]:
movies = pd.read_csv("../data/movies.csv")

In [3]:
ratings = pd.read_csv("../data/ratings.csv")

In [4]:
movie_ratings = ratings.merge(movies, on="movieId")

In [5]:
n_reviews = movie_ratings["timestamp"].value_counts().reset_index()

In [6]:
timestamp_l = 0
timestamp_r = n_reviews["timestamp"].max()
total_sum = n_reviews["count"].sum()
fraction = 0.8
while timestamp_l + 1 < timestamp_r:
    timestamp_mid = (timestamp_l + timestamp_r) // 2
    current_sum = n_reviews[n_reviews["timestamp"] < timestamp_mid]["count"].sum()
    if current_sum < (fraction * total_sum):
        timestamp_l = timestamp_mid
    else:
        timestamp_r = timestamp_mid
print(f"{fraction} of data has timestamp less than {timestamp_l}")

0.8 of data has timestamp less than 1466837397


# If Train and Validation
## Train Time Margin:
timestamp < 1421944985
## Test Time Margin:
1421944985 <= timestamp < 1515033908
## Validation Time Margin:
1515033908 <= timestamp
# If just Validation
## Train Time Margin:
timestamp < 1466837397
## Validation Time Margin:
1466837397 <= timestamp